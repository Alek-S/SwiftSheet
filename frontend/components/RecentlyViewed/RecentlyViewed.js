import React from 'react';
import styled from 'styled-components';
import { Card } from '../../defaultStyle';
import { getHistory } from '../../utils/history';
import { Link } from 'react-router-dom';

const RecentlyViewed = () => {
	const listHistory = getHistory()
		? getHistory().map((item, i) => (
				<Link to={`/${item}`} key={`${item.toString()}-${i}`}>
					<li>{item}</li>
				</Link>
		  ))
		: null;

	return (
		<StyledCard>
			<header>
				<h2>Recently Viewed Sheets</h2>
			</header>
			{listHistory ? (
				<ul>{listHistory}</ul>
			) : (
				<p className="no-items">No recently viewed items</p>
			)}
		</StyledCard>
	);
};

const StyledCard = styled(Card)`
	padding-left: 0px;
	padding-right: 0px;
	padding-top: 0;
	overflow: hidden;
	width: 90%;
	max-width: 600px;

	header {
		background: ${props => props.theme.gradient.greenBlue};
		padding-top: 1rem;
		padding-bottom: 1rem;
		width: 100%;
	}

	h2 {
		font-weight: 400;
		text-align: center;
		color: white;
		font-size: 1.1rem;
	}

	.no-items {
		font-style: italic;
		padding-top: 3rem;
		padding-bottom: 3rem;
		text-align: center;
		font-size: 0.9rem;
		color: ${props => props.theme.color.text};
	}

	a {
		text-decoration: none;
		color: ${props => props.theme.color.text};
	}

	li {
		font-size: 0.9rem;
		color: ${props => props.theme.color.text};
		padding-left: 1.5rem;
		padding-top: 1rem;
		padding-bottom: 1rem;
		margin: 1rem;
		border-radius: 8px;
		border: 1px solid ${props => props.theme.color.border};
		text-align: center;
		transition: all 0.25s;
	}
	li:hover {
		border: 1px solid ${props => props.theme.color.aqua};
		background-color: ${props => props.theme.color.background};
	}
`;

export default RecentlyViewed;
