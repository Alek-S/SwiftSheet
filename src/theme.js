/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
