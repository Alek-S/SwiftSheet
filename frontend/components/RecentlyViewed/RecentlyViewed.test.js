import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';

import RecentlyViewed, { StyledCard } from './RecentlyViewed';
import theme from '../../theme';

describe('RecentlyViewed', () => {
	test('matches the snapshot', () => {
		const tree = shallowWithTheme(<RecentlyViewed />, theme).dive();
		expect(toJson(tree)).toMatchSnapshot();
	});

	test('StyledCard matches the snapshot', () => {
		const tree = renderer.create(<StyledCard theme={theme} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
