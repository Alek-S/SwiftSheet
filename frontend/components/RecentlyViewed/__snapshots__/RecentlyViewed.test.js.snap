// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentlyViewed StyledCard matches the snapshot 1`] = `
.c0 {
  background-color: white;
  color: #404040;
  font-family: 'Quicksand',sans-serif;
  font-size: 0.9em;
  border-radius: 8px;
  box-shadow: 0px 0px 30px -5px rgba(0,0,0,0.15);
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0;
  overflow: hidden;
  width: 90%;
  max-width: 600px;
}

.c0 input {
  outline: none;
  color: #404040;
  font-family: 'Quicksand',sans-serif;
  background-color: #F1F3F4;
  border: 1px solid #F1F3F4;
  border-radius: 5rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 input::-webkit-input-placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c0 input::-moz-placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c0 input:-ms-input-placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c0 input::placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c0 input:hover {
  border: solid 1px #d9d9d9;
}

.c0 input:focus,
.c0 input:active {
  border: solid 1px #46C8F5;
}

.c0 input.true {
  border: 1px solid #ED5050;
}

.c0 header {
  background: linear-gradient(160deg,#57DACB 0%,#55A2ED 100%);
  padding-top: 1rem;
  padding-bottom: 1rem;
  width: 100%;
}

.c0 h2 {
  font-weight: 400;
  text-align: center;
  color: white;
  font-size: 1.1rem;
}

.c0 .no-items {
  font-style: italic;
  padding-top: 3rem;
  padding-bottom: 3rem;
  text-align: center;
  font-size: 0.9rem;
  color: #404040;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #404040;
}

.c0 li {
  font-size: 0.9rem;
  color: #404040;
  padding-left: 1.5rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  margin: 1rem;
  border-radius: 8px;
  border: 1px solid #d9d9d9;
  text-align: center;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.c0 li:hover {
  border: 1px solid #68F6BA;
  background-color: #fafafa;
}

<div
  className="c0"
/>
`;

exports[`RecentlyViewed matches the snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "RecentlyViewed__StyledCard-o9muk8-0",
        "isStatic": false,
        "rules": Array [
          "background-color:white;color:",
          [Function],
          ";font-family:",
          [Function],
          ";font-size:0.9em;",
          "border-radius:8px;box-shadow:",
          [Function],
          ";input{outline:none;color:",
          [Function],
          ";font-family:",
          [Function],
          ";background-color:",
          [Function],
          ";border:1px solid ",
          [Function],
          ";border-radius:5rem;transition:all 0.3s;::placeholder{font-size:1rem;font-weight:400;font-style:italic;font-family:",
          [Function],
          ";color:",
          [Function],
          ";}&:hover{border:solid 1px ",
          [Function],
          ";}&:focus,&:active{border:solid 1px ",
          [Function],
          ";}&.true{border:1px solid ",
          [Function],
          ";}}",
          "padding-left:0px;padding-right:0px;padding-top:0;overflow:hidden;width:90%;max-width:600px;header{background:",
          [Function],
          ";padding-top:1rem;padding-bottom:1rem;width:100%;}h2{font-weight:400;text-align:center;color:white;font-size:1.1rem;}.no-items{font-style:italic;padding-top:3rem;padding-bottom:3rem;text-align:center;font-size:0.9rem;color:",
          [Function],
          ";}a{text-decoration:none;color:",
          [Function],
          ";}li{font-size:0.9rem;color:",
          [Function],
          ";padding-left:1.5rem;padding-top:1rem;padding-bottom:1rem;margin:1rem;border-radius:8px;border:1px solid ",
          [Function],
          ";text-align:center;transition:all 0.25s;}li:hover{border:1px solid ",
          [Function],
          ";background-color:",
          [Function],
          ";}",
        ],
      },
      "displayName": "RecentlyViewed__StyledCard",
      "foldedComponentIds": Array [
        "defaultStyle-g21vfe-0",
        "defaultStyle__Card-g21vfe-1",
      ],
      "render": [Function],
      "styledComponentId": "RecentlyViewed__StyledCard-o9muk8-0",
      "target": "div",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <header>
    <h2>
      Recently Viewed Sheets
    </h2>
  </header>
  <p
    className="no-items"
  >
    No recently viewed items
  </p>
</StyledComponent>
`;
